### ISSUE TITLE
Shorter decoding segments when using server/x86

### ISSUE THREAD

**Issue by OleguerCanal** [wenet-e2e/wenet#1236]
**Problem:**
I was exploring word-level confidences using the transformer decoders and I noticed the following pattern in the "forward-in-time" decoder:The last half part of the logits sequences the ctc-guessed probabilities where very low in favor of <eos> and other wrong guesses. I am not sure about it, but I suspect this error might have tot do with the sequence length difference when training (around 10 seconds) and inference (around 30 seconds).
I believe that this is a big problem because the attention re-scoring values might be all wrong.

**Describe the solution you'd like**
I think an easy-to-test solution would be to do inference in shorter segments. While we loose context, we don't have this <eos> issue. However, I am not sure how to change the segment length.

**Comment by OleguerCanal** [comment]
The rescoring accuracy is also wrong because it is just the addition of each word's probability. Which at some point becomes erroneous, so the rescoring is not working as expected in the paper.
